worlds/mm2/rules.py

```diff
@@ -0,0 +1,296 @@
+                    used = ceil(boss_health[boss] / boss_damage[wp])
+                    weapon_energy[wp] -= weapon_costs[wp] * used
+                    boss_health[boss] = 0
+                elif highest <= 0:
+                    # we are out of weapons that can actually damage the boss
+                    # so find the weapon that has the most uses, and apply that as an additional weakness
+                    # it should be impossible to be out of energy, simply because even if every boss took 1 from
+                    # Quick Boomerang and no other, it would only be 28 off from defeating all 9, which Metal Blade should
+                    # be able to cover
+                    wp, max_uses = max((weapon, weapon_energy[weapon] // weapon_costs[weapon]) for weapon in weapon_weight
+                                       if weapon != 0)
+                    world.weapon_damage[wp][boss] = minimum_weakness_requirement[wp]
+                    used = min(int(weapon_energy[wp] // weapon_costs[wp]),
+                               ceil(boss_health[boss] // minimum_weakness_requirement[wp]))
+                    weapon_energy[wp] -= weapon_costs[wp] * used
+                    boss_health[boss] -= int(used * minimum_weakness_requirement[wp])
+                    weapon_weight.pop(wp)
+                else:
+                    # drain the weapon and continue
+                    boss_health[boss] -= int(uses * boss_damage[wp])
+                    weapon_energy[wp] -= weapon_costs[wp] * uses
+                    weapon_weight.pop(wp)
+
+    for i, boss_locations in zip(range(14), [
+        heat_man_locations,
+        air_man_locations,
+        wood_man_locations,
+        bubble_man_locations,
+        quick_man_locations,
+        flash_man_locations,
+        metal_man_locations,
+        crash_man_locations,
+        wily_1_locations,
+        wily_2_locations,
+        wily_3_locations,
+        wily_4_locations,
+        wily_5_locations,
+        wily_6_locations
+    ]):
+        if world.weapon_damage[0][i] > 0:
+            continue  # this can always be in logic
+        weapons = []
+        for weapon in range(1, 9):
+            if world.weapon_damage[weapon][i] > 0:
+                if world.weapon_damage[weapon][i] < minimum_weakness_requirement[weapon]:
+                    continue  # Atomic Fire can only be considered logical for bosses it can kill in 2 hits
+                weapons.append(weapons_to_name[weapon])
+        if not weapons:
+            raise Exception(f"Attempted to have boss {i} with no weakness! Seed: {world.multiworld.seed}")
+        for location in boss_locations:
+            if i == 12:
+                add_rule(world.multiworld.get_location(location, world.player),
+                         lambda state, weps=tuple(weapons): state.has_all(weps, world.player))
+                # TODO: when has_list gets added, check for a subset of possible weaknesses
```

Comment:
It's a little more complex than just swapping out the rule, since I have to restructure the rom patching to work alongside the change (as it's the rom change that actually makes this notable). Probably can do it now though.

Reply:
I'd definitely understand if you wanted to do it in a later PR. This was mostly a non-serious comment which I really should have indicated better, mb